hoodie-client/src/main/java/com/uber/hoodie/func/BufferedIterator.java

```diff
@@ -82,13 +80,16 @@
   // indicates number of samples collected so far.
   private long numSamples = 0;
   // next record to be read from buffer.
-  private BufferedIteratorPayload<T> nextRecord;
+  private AbstractBufferedIteratorPayload<T,R> nextRecord;
+  // Function to transform the input payload to the expected output payload
+  private PayloadFunction<T, R> payloadCreator;
 
   public BufferedIterator(final Iterator<T> iterator, final long bufferMemoryLimit,
-      final Schema schema) {
+      final Schema schema, final PayloadFunction<T, R> payloadFunc) {
```

Comment:
Minor: Rename PayloadFunction to BufferedIteratorPayloadFactory as it is used only in the context of BufferedIterator